eng/common/pipelines/templates/steps/detect-api-changes.yml

```diff
@@ -3,6 +3,12 @@ parameters:
   Artifacts: []
 
 steps:
+  - pwsh: |
+      $apiChangeDetectRequestUrl = "https://apiview.dev/PullRequest/DetectApiChanges"
+      echo "##vso[task.setvariable variable=ApiChangeDetectRequestUrl]$apiChangeDetectRequestUrl"
+    displayName: "Set API change detect request URL"
+    condition: eq(variables['ApiChangeDetectRequestUrl'], '')
```

Comment:
Have you tried setting this as the default in a `variables` block in this file? It would be simpler.

```
variables:
  ApiChangeDetectRequestUrl: <value that gets overwritten>
```

Also, how should a pipeline template caller override this variable? Would it be easier to use a devops parameter and default it instead?

```
parameters:
  ApiChangeDetectRequestUrl: <url>
```

Reply:
Why didn't we override existing param to support different URL from caller YAML:

Caller pipeline yaml is same in test as well as main repo since it is a copy of azure sdk language repos. Even if we modify caller pipeline then it will get overridden with changes from main branch.

pipeline doesn't have any step level variable. We will need to set this as job level variable for all languages in caller side. Devops is failing with incorrect yaml when variables is added in this block.

Caller pipeline can setup pipeline queue time variable whenever we want to test manually against different instance. This change is to support testing API change detection using non prod APIView instance for demo or for manual testing.